libs/dal/src/repositories/preferences/preferences.entity.ts

```diff
@@ -0,0 +1,54 @@
+import { ChannelTypeEnum } from '@novu/shared';
+import type { OrganizationId } from '../organization';
+import type { EnvironmentId } from '../environment';
+import type { SubscriberId } from '../subscriber';
+import { UserId } from '../user';
+import { ChangePropsValueType } from '../../types';
+
+type PreferenceOptions = {
+  defaultValue: boolean;
+  readOnly: boolean;
+};
+
+type WorkflowOptionsPreferences = {
+  workflow: PreferenceOptions;
+  channels: Record<ChannelTypeEnum, PreferenceOptions>;
+};
+
+export enum PreferencesTypeEnum {
+  SUBSCRIBER_GLOBAL = 'SUBSCRIBER_GLOBAL',
+  SUBSCRIBER_WORKFLOW = 'SUBSCRIBER_WORKFLOW',
+  USER_WORKFLOW = 'USER_WORKFLOW',
+  CODE_FIRST_WORKFLOW = 'CODE_FIRST_WORKFLOW',
```

Comment:
The naming is semantic, but I think we can do better to abstract away from the V2 experience. How about the following?

```suggestion
  WORKFLOW_RESOURCE = 'WORKFLOW_RESOURCE',
```